src/ServiceControl.UnitTests/Recoverability/HeaderFilterTests.cs

```diff
@@ -0,0 +1,57 @@
+﻿namespace ServiceControl.UnitTests.Recoverability
+{
+    using System.Collections.Generic;
+    using NUnit.Framework;
+    using Particular.Approvals;
+    using ServiceControl.Recoverability;
+
+    [TestFixture]
+    public class HeaderFilterTests
+    {
+        [Test]
+        public void RemovesHeaders()
+        {
+            var headers = new Dictionary<string, string>
+            {
+                { "NServiceBus.MessageId", "caf68027-acce-4260-8442-ac6500e46afc" },
+                { "NServiceBus.MessageIntent", "Publish" },
+                { "NServiceBus.RelatedTo", "2e6c6c9c-c4cb-474e-98b8-ac6500e46a0c" },
+                { "NServiceBus.ConversationId", "37c405c8-e4e4-4873-8eb4-ac6500e46621" },
+                { "NServiceBus.CorrelationId", "6e449174-6f77-4da4-b9c0-ac6500e46621" },
+                { "NServiceBus.OriginatingMachine", "MACHINE"},
+                { "NServiceBus.OriginatingEndpoint", "PurchaseOrderService.1.0" },
+                { "$.diagnostics.originating.hostid", "4f8138bdb0421ffe1ceaee86e9145721" },
+                { "NServiceBus.OriginatingSagaId", "9e0d2f01-e903-481a-b272-ac6500e46715" },
+                { "NServiceBus.OriginatingSagaType", "PowerSupplyPurchaseOrderService.PurchaseOrderSaga, PowerSupplyPurchaseOrderService, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null" },
+                { "NServiceBus.ReplyToAddress", "PowerSupplyPurchaseOrderService.1.0@[dbo]@[Market.NServiceBus.Prod]" },
+                { "NServiceBus.ContentType", "application/json" },
+                { "NServiceBus.EnclosedMessageTypes", "PowerSupplyPurchaseOrderService.ApiModels.Events.v1_0.PowerSupplyDebtorBlacklistCheckCompleted, PowerSupplyOrderService.ApiModels, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null" },
+                { "NServiceBus.Version", "7.1.0" },
+                { "NServiceBus.TimeSent", "2020-10-31 13:59:26:479745 Z" },
+                { "NServiceBus.Retries.Timestamp", "2020-10-31 13:56:55:359877 Z" },
```

Comment:
according to your comment on https://github.com/Particular/ServiceControl/issues/2259, shouldn't the retries header be removed too?

Reply:
It should. Let me double check. Good catch

Reply:
Ah. Hehe he. It is. Check the approval file. This is the arrange